### PR TITLE
Enable story player for search and archives

### LINKED ISSUE #3166: Display Stories in Story Player in archives
## Feature Description

On a WordPress site, stories can appear in search results and eventually archives (#3063).

However, the look & feel will be very dire, as only the excerpt is displayed. This is an opportunity to display the actual story in an `<amp-story-player>`.

## Alternatives Considered

We could just leave as-is, but that might not be desired.

## Additional Context

Status quo:

Screenshot 2020-07-13 at 15 07 35

---

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Implementation Brief

* Filter story content on non-single views to display them in an `<amp-story-player>`

### PR DESCRIPTION
**Pull request by spacedmonkey**
## Summary

Enable embedded player for search or custom post type archives.

## Relevant Technical Choices

I added filters on both the_content and the excerpt as you never know which one is used in template. 
I added max php int, to avoid conflicts with other plugins. 

## Testing Instructions

---

Fixes #3166

### CODE REVIEW

File: includes/Story_Post_Type.php [hunk 1]
```diff
@@ -614,4 +620,33 @@ public function add_to_jetpack_sitemap( $post_types ) {
...
+
+		if ( $post instanceof WP_Post && self::POST_TYPE_SLUG === $post->post_type ) {
+			$story = new Story();
+			$story->load_from_post( $post );
+
+			$embed   = new Embed( $story );
+			$content = $embed->render();
+		}
+
+		return $content;
```
Comment (swissspidy) [inline includes/Story_Post_Type.php:677]: Nit, but could do bail early for consistency:

```suggestion
		if ( ! $post instanceof WP_Post || self::POST_TYPE_SLUG !== $post->post_type ) {
			return $content;
		}

		$story = new Story();
		$story->load_from_post( $post );

		$embed   = new Embed( $story );
		return $embed->render();
```

File: includes/Story_Post_Type.php [hunk 2]
```diff
@@ -614,4 +620,33 @@ public function add_to_jetpack_sitemap( $post_types ) {
...
+	 * @return string
+	 */
+	public function embed_player( $content ) {
+		$post = get_post();
+
+		if ( is_feed() ) {
+			return $content;
+		}
+
+		if ( ! is_search() && ! is_post_type_archive() ) {
```
Comment (swissspidy) [inline includes/Story_Post_Type.php]:
```suggestion
		if ( ! is_search() && ! is_post_type_archive( self::POST_TYPE_SLUG ) ) {
```

**Review by swissspidy — CHANGES_REQUESTED** [review]
2 nits & a merge conflict

(sounds like a movie title 🤔 )

otherwise LGTM! 👍

File: includes/Story_Post_Type.php (hunk 1 above)
Comment (spacedmonkey) [reply includes/Story_Post_Type.php:677]: You bail early to start a large amount of processing this only stop 4 lines. It also means that I would have to write a test for another post to check this work. I don't see the point.

File: includes/Story_Post_Type.php (hunk 2 above)
Comment (spacedmonkey) [reply includes/Story_Post_Type.php]: Good catch!

File: includes/Story_Post_Type.php (hunk 1 above)
Comment (spacedmonkey) [reply includes/Story_Post_Type.php:677]: Single return functions are much cleaner.

Comment (swissspidy) [reply includes/Story_Post_Type.php:677]: The branch coverage would be the same, so I don't see how this would require another test, but alas 🤷

**Review by swissspidy — APPROVED** [review]

**Pull request merged**